aiven/client/argx.py

```diff
@@ -167,13 +150,13 @@ def save(self) -> None:
             jsonlib.dump(self, fp, sort_keys=True, indent=4)
 
 
-class CommandLineTool:  # pylint: disable=old-style-class
+class CommandLineTool:
     config: Config
 
-    def __init__(self, name: str, parser: Optional[argparse.ArgumentParser] = None):
+    def __init__(self, name: str, parser: argparse.ArgumentParser | None = None):
```

Comment:
`T1 | T2`  syntax is only in Python 3.10+ ... we can't get rid of it yet with 3.8 + 3.9 support :-(

Reply:
It's possible because of https://peps.python.org/pep-0563/. Also, we have testing suite which pass.
![image](https://github.com/aiven/aiven-client/assets/3449635/249bfdc9-520f-430d-81a5-1ed66b460b81)

Reply:
https://github.com/python/mypy/issues/9610

Reply:
I guess it works if we're using "recent enough" version of `mypy` ... and not using something like pydantic that actually looks and tries to resolve the type hints, or else it would result in something this:
```python
>>> from aiven.client.argx import name_to_cmd_parts
>>> from typing import get_type_hints
>>> get_type_hints(name_to_cmd_parts)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.8/typing.py", line 1264, in get_type_hints
    value = _eval_type(value, globalns, localns)
  File "/usr/lib64/python3.8/typing.py", line 270, in _eval_type
    return t._evaluate(globalns, localns)
  File "/usr/lib64/python3.8/typing.py", line 518, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
TypeError: 'type' object is not subscriptable

Reply:
> I guess it works if we're using "recent enough" version of mypy

Mypy version pinned in dependencies supports this, so I see no issue here.

> and not using something like pydantic that actually looks and tries to resolve the type hints

I believe it's not that important.